src/cache.py

```
@@ -0,0 +1,103 @@
+    data = doc.to_dict() or {}
+    parts = data.get("participants")
+    if not parts:
+        from database import ChatDBClient
+        parts = ChatDBClient().get_participants(gid)
+        groups.document(gid).set({"participants": parts}, merge=True)
+    return parts
+
+class RedisCache:
+    def __init__(self, red: aioredis.Redis):
+        self.red = red
+    async def get_user(self, uid: str) -> Dict[str, Any]:
```

Flake8 (7.2.0)

[error] expected 1 blank line, found 0

(E301)